spec/controllers/answer_tags_controller_spec.rb

```diff
@@ -117,4 +127,46 @@
       end
     end
   end
+
+
+  #To allow creation if not existing and simultaneously updating the new answer tag.
+  #params: answer_id (answer id mapping to which tag is being created)
+  #params: tag_prompt_deployment_id (tag_prompt id mapping to which tag is being created)
+  #params: value (new value to be updated)
+
+  describe '#create_edit' do
+    context 'when student tries to create or update the answer tags' do
+      before(:each) do
+        controller.request.session[:user] = student
+      end
+
+      it 'add entry if not existing and update the old value by new value provided as param' do
+        params = {answer_id: answer.id,tag_prompt_deployment_id: tag_deploy.id,value: 0}
```

Comment:
The `value` field in `AnswerTag` is a `varchar(255)` so it should be a string rather than an integer.
```suggestion
        params = {answer_id: answer.id,tag_prompt_deployment_id: tag_deploy.id,value: "0"}
```

Reply:
Thanks for the suggestion.